fdbserver/ClusterController.actor.cpp

```diff
@@ -2719,6 +2719,71 @@ class ClusterControllerData {
 		return idUsed;
 	}
 
+    // Updates work health signals in `workerHealth` based on `req`.
+    void updateWorkerHealth(const UpdateWorkerHealthRequest& req) {
+        std::string degradedPeersString;
+        for (int i = 0; i < req.degradedPeers.size(); ++i) {
+            degradedPeersString += i == 0 ? "" : " " + req.degradedPeers[i].toString();
+        }
+        TraceEvent("ClusterControllerUpdateWorkerHealth")
+            .detail("WorkerAddress", req.address)
+            .detail("DegradedPeers", degradedPeersString);
+
+        double currentTime = now();
+        auto it = workerHealth.find(req.address);
+
+        // Current `workerHealth` doesn't have any information about the incoming worker. Add the worker into
+        // `workerHealth`.
+        if (it == workerHealth.end()) {
+            if (req.degradedPeers.empty()) {
+                return;
+            }
+
+            workerHealth[req.address] = {};
+            for (const auto& degradedPeer : req.degradedPeers) {
+                workerHealth[req.address].degradedPeers[degradedPeer] = { currentTime, currentTime };
+            }
+
+            return;
+        }
+
+        // The incoming worker already exists in `workerHealth`.
+
+        // `req.degradedPeers` contains the latest peer performance view from the worker. Clear the worker if the
+        // requested worker doesn't see any degraded peers.
```

Comment:
This can be moved to before line 2735. If `req.degradedPeers.empty()`, we erase req.address, which is a safe operation even the key doesn't exist, and return immediately. Then line 2738-2740 can be removed.

Reply:
Done.